tests/profiler/bob_user_workflow_verbose_profiler_config.yml

```diff
@@ -0,0 +1,216 @@
+        max_value: $parameter.my_stdev_of_means_parameter_for_batches_in_previous_full_year.stdev * 1.2
+        # Alternatively, we could have used a MultiBatchParameterBuilder and processed the means in the ExpectationConfigurationBuilder as in my_rule_for_ids
+        # min_value: stdev($parameter.my_stdev_of_means_parameter_for_batches_in_previous_full_year.means) * 0.8
+        # max_value: stdev($parameter.my_stdev_of_means_parameter_for_batches_in_previous_full_year.means) * 1.2
+        # Note: this would need us to add `stdev()` to our GE language parsing these evaluations
+
+  my_rule_for_numeric_data:
+    # This rule creates several expectations for all numeric column domains. There is some overlap with the my_rule_for_amounts rule
+    # Rationale: I want alerts when my current numeric data varies from prior data (within some bounds, based on a confidence level)
+    class_name: Rule
+    module_name: great_expectations.profiler.rule.rule
+    domain_builder:
+      # Build list of domains from all numeric column domains by checking column storage type using the configured execution engine
+      class_name: SimpleSemanticTypeColumnDomainBuilder
+      module_name: great_expectations.profiler.domain_builder.simple_semantic_type_domain_builder
+      semantic_types:
+        - numeric
+    parameter_builders:
+      # Note: my_column_mean and my_column_min are added as parameters but unused in expectation_configuration_builders - this is OK, we should still accept this as a valid Profiler configuration.
+      # TODO: Maybe in the future we should warn users if there are unused ParameterBuilders in a Profiler configuration.
+      - name: my_column_mean
+        class_name: MetricParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.metric_parameter_builder
+        metric_name: column.mean
+        metric_domain_kwargs: $domain.domain_kwargs
+      - name: my_column_min
+        class_name: MetricParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.metric_parameter_builder
+        metric_name: column.min
+        metric_domain_kwargs: $domain.domain_kwargs
+      - name: my_column_max
+        class_name: MetricParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.metric_parameter_builder
+        metric_name: column.max
+        metric_domain_kwargs: $domain.domain_kwargs
+
+      - name: false_positive_threshold_min
+        class_name: MultiBatchBootstrappedMetricDistributionParameterBuilder
+        module_name: great_expectations.profiler.parameter_builder.multi_batch_bootstrapped_metric_distribution_parameter_builder
+        # TODO: Not sure I understand the comment here about unioning - why would we want to add the batches from the call to profile? Wouldn't we expect that this parameter is calculated just from the explicitly defined batch_request below? (if this is useful, maybe there are two modes)
```

Comment:
Just re-iterating. What batches are available here? I could imagine that rather than passing a batch into the profiler, we pass in a DataConnector. Then we review all of the BatchRequests contained in the config, and instantiate a validator containing all of them

Reply:
Interesting - so change the `Profiler.profile()` method to take a data connector instead... I think that makes sense and jibes with James' comment about unioning batches. Currently we aren't doing that yet.

Reply:
That's what I'm thinking. Otherwise I can see a lot of situation where a user tries to set up batches which aren't passed in. If we're allowing BatchRequest-ish things within the config, I think this would make sense as well - otherwise the BatchRequest would only work against BatchRequests already contained with the Validator, which feels very limited.

Reply:
This may be silly - but would we create a new validator within the builder to handle the specific BatchRequest? I guess that would defeat the purpose of the computation tree and there is a chance we would re-compute metrics (unless we were storing them somewhere accessible to any validators)

Reply:
@talagluck @anthonyburdi I would love to discuss use cases at length.  The way I see it now, no component of `DataSource`, including `DataConnector` must be surfaced as parameters the API level.  All the smarts should be expressed via `BatchRequest` -- and it should be sufficient to express the querying for any bathes.  Should we find a limitation, then we would enrich the `BatchRequest` capabilities, not expose its processing internals.  I believe that if we describe use cases and the requirements for `Batch` objects that should be found, then we can figure out whether or not the existing `BatchRequest` expressivity is sufficient.  Thanks.